src/wagtailmedia/widgets.py

```diff
@@ -34,9 +34,23 @@ class AdminMediaChooser(AdminChooser):
     link_to_chosen_text = _("Edit this media item")
 
     def __init__(self, **kwargs):
-        super(AdminMediaChooser, self).__init__(**kwargs)
+        media_type = kwargs.pop("media_type", None)
+        super().__init__(**kwargs)
         self.media_model = get_media_model()
 
+        if media_type == "audio":
```

Comment:
Is it possible to create sub-classes for each media type, for consistency with blocks?

Also, do we want to be able to support custom media types? (such as svg, 3d models, etc)

Reply:
good suggestion! I did the split for the blocks quite late, but creating custom widget classes means we'd have a pattern for other media types, should we want to support them

Reply:
done now